### PR TITLE
Honor proxy usage

### PR DESCRIPTION
**Pull request by baude**
If HTTP[S]_PROXY is defined, honor it in python requests usage
as well as pass it on to skopeo.

### CODE REVIEW

**Comment by baude** [comment]
[(?) Atomic CLI support for HTTP/HTTPS_PROXY env vars](https://trello.com/c/qa3obU4v/353-atomic-cli-support-for-http-https-proxy-env-vars)

File: Atomic/push.py [hunk 1]
```diff
@@ -69,6 +69,8 @@ def cli(subparser):
                               "use an alternate user's GPG keyring for signing. "
                               "Useful when running with sudo, "
                               "e.g. set to '~/.gnupg'."))
+    pushp.add_argument("--insecure", dest="insecure", default=False,
```
Comment (giuseppe) [inline Atomic/push.py]: should this new option be added to bash autocomplete and to the documentation?

**Comment by rhatdan** [comment]
It seems that there are multiple pull requests combined here.  Can we separate them out.  I agree with giuseppe on the need for man page and command completion.

File: Atomic/util.py [hunk 2]
```diff
@@ -124,7 +123,8 @@ def subp(cmd, cwd=None, newline=False):
     proc = subprocess.Popen(cmd, cwd=cwd,
                             stdout=subprocess.PIPE,
                             stderr=subprocess.PIPE, close_fds=True,
-                            universal_newlines=newline)
+                            universal_newlines=newline,
+                            env=os.environ)
```
Comment (rhatdan) [inline Atomic/util.py]: What do these have to do with the --insecure flag?

Comment (baude) [inline Atomic/util.py]: When we call skopeo, we need to pass the proxy information which is done via environment variables.

File: Atomic/push.py (hunk 1 above)
Comment (baude) [inline Atomic/push.py]: It is already in the bash completion, I'll add to the docs.

**Comment by baude** [comment]
@rhatdan there are not multiple requests.  This PR covers proxy work and that we need to be able to allow the user to indicate if the registry is insecure because we cannot authoritatively be sure.

**Comment by baude** [comment]
@rhatdan I added the ability to read atomic.conf.  Ready for re-review

**Comment by baude** [comment]
bot, retest this please

**Comment by ddarrah** [comment]
LGTM - this will really help simplify CI.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ c0a844f

**Pull request closed without merging**